compiler/test/dotty/tools/dotc/ParallelTesting.scala

```diff
@@ -1021,11 +1022,17 @@ trait ParallelTesting { self =>
       if (f.isDirectory) f.listFiles.flatMap(flatten)
       else Array(f)
 
+    val files = {
+      // Deterministically randomises compilation order
+      Random.setSeed(seed)
+      Random.shuffle(flatten(sourceDir).toList).toArray
```

Comment:
Some other thread could mess with the global instance of `Random`. You should use a new instance of `Random` in here.